project/Build.scala

```diff
@@ -582,12 +579,23 @@ object Build {
       base = file("tools/js"),
       settings = myScalaJSSettings ++ commonToolsSettings ++ Seq(
           crossVersion := ScalaJSCrossVersion.binary,
-          resources in Test <+= jsTestDefinitions in (testSuite, Test),
-          jsDependencies += {
-            val artifact =
-              (artifactPath in (testSuite, Test, jsTestDefinitions)).value
-            ProvidedJS / artifact.getName % "test"
-          }
+          resourceGenerators in Test <+= Def.task {
+            val base = (resourceManaged in Compile).value
+            IO.createDirectory(base)
+            val outFile = base / "js-test-definitions.js"
+
+            val writer = new BufferedWriter(new FileWriter(outFile))
+            try {
+              org.scalajs.sbtplugin.HTMLRunnerTemplate.renderTestDefinitions(
```

Comment:
Yeah, probably. I'm actually still considering whether we should write the JS test definitions separately. Then we could also make the `TestDetector` a first-class citizen. But we should probably wait until somebody asks us to.